ibis/streamlit/__init__.py

```diff
@@ -0,0 +1,27 @@
+from __future__ import annotations
+
+from typing import Any
+
+from streamlit.connections import ExperimentalBaseConnection
+from streamlit.runtime.caching import cache_data
+
+import ibis
+from ibis.backends.base import BaseBackend
+
+__all__ = ["IbisConnection"]
+
+
+class IbisConnection(ExperimentalBaseConnection[BaseBackend]):
+    def _connect(self, **kwargs) -> BaseBackend:
+        kwargs.update(self._secrets.to_dict())
+        kwargs.pop("type", None)
```

Comment:
What is the `type` option we're ignoring here?

Reply:
It's the `type` argument here:

```python
con = st.experimental_connection("ch", type=IbisConnection)
```

We don't want to pass this to `ibis.connect` since `ibis.connect` doesn't know what to do with `type`.

Reply:
Should we file this as an upstream bug? Since AFAICT `experimental_connection` creates a connection by calling `type._connect(**kwargs)`, I'd expect any implementation to want to ignore the `type` kwarg.

Reply:
It's possible that was fixed since the time I wrote this a month or two ago. Let me confirm and then I'll file a bug if it's still an issue.